File: jetpack_java/src/main/java/com/example/jetpack_java/sample_04_databinding/ui/DataBindingListActivity.java
/*
 * Copyright 2018-2020 KunMinX
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.example.jetpack_java.sample_04_databinding.ui;

import android.content.Intent;
import android.os.Bundle;

import androidx.databinding.DataBindingUtil;
import androidx.recyclerview.widget.RecyclerView;

import com.example.jetpack_java.R;
import com.example.jetpack_java.databinding.ActivityDatabindingListBinding;
import com.example.jetpack_java.databinding.AdapterDatabindingMomentBinding;
import com.example.jetpack_java.sample_01_lifecycles.data.Configs;
import com.example.jetpack_java.sample_04_databinding.data.bean.Moment;
import com.example.jetpack_java.sample_04_databinding.ui.state.ListViewModel;
import com.kunminx.architecture.ui.BaseActivity;
import com.kunminx.architecture.ui.adapter.SimpleBindingAdapter;

/**
 * Create by KunMinX at 19/10/16
 */

public class DataBindingListActivity extends BaseActivity {

    private ListViewModel mListViewModel;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        mListViewModel = getActivityViewModel(ListViewModel.class);

        ActivityDatabindingListBinding binding = DataBindingUtil.setContentView(this, R.layout.activity_databinding_list);

        binding.setVm(mListViewModel);

        binding.setAdapter(new SimpleBindingAdapter<Moment, AdapterDatabindingMomentBinding>(getApplicationContext(), R.layout.adapter_databinding_moment) {
            @Override
            protected void onSimpleBindItem(AdapterDatabindingMomentBinding binding, Moment moment, RecyclerView.ViewHolder holder) {
                Intent intent = new Intent(DataBindingListActivity.this, DataBindingDetailActivity.class);
                intent.putExtra(Configs.THIS_MOMENT, moment);
                startActivity(intent);
            }
        });

        mListViewModel.getListMutableLiveData().observe(this, moments -> {
            mListViewModel.list.setValue(moments);
        });

        mListViewModel.requestList();
    }

}

File: jetpack_kotlin/src/main/java/com/flywith24/jetpack_kotlin/sample_04_databinding/DataBindingListActivity.kt
package com.flywith24.jetpack_kotlin.sample_04_databinding

import android.content.Intent
import android.os.Bundle
import androidx.activity.viewModels
import androidx.databinding.DataBindingUtil
import androidx.lifecycle.observe
import com.flywith24.jetpack_kotlin.R
import com.flywith24.jetpack_kotlin.common_data.bean.Moment
import com.flywith24.jetpack_kotlin.common_ui.adapter.MomentAdapter
import com.flywith24.jetpack_kotlin.databinding.KotlinActivityListDatabindingBinding
import com.flywith24.jetpack_kotlin.sample_04_databinding.state.ListViewModel
import com.kunminx.architecture.ui.BaseActivity

/**
 * @author Flywith24
 * @date   2020/5/31
 * time   23:29
 * description
 */
class DataBindingListActivity : BaseActivity(), MomentAdapter.OnItemClickListener {
    private val mListViewModel by viewModels<ListViewModel>()
    private val mAdapter by lazy { MomentAdapter(this) }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        val binding = DataBindingUtil.setContentView<KotlinActivityListDatabindingBinding>(this, R.layout.kotlin_activity_list_databinding)
        binding.lifecycleOwner = this
        binding.vm = mListViewModel
        binding.click = ClickProxy()
        binding.adapter = mAdapter

        mListViewModel.getListMutableLiveData().observe(this) { moments ->
            mListViewModel.list.value = moments
        }

        mListViewModel.requestList()
    }

    inner class ClickProxy {
        fun fabClick() {
            startActivity(Intent(this@DataBindingListActivity, DataBindingEditorActivity::class.java))
        }
    }

    override fun onItemClick(moment: Moment) {

    }
}